egui/src/widgets/button.rs

```diff
@@ -165,16 +189,30 @@ impl Widget for Button {
         }
         desired_size = desired_size.at_least(min_size);
 
+        if let Some(image) = image {
+            let image_padding = image.size() + 4.0 * button_padding;
+            desired_size.x += image_padding.x;
+            desired_size.y = image_padding.y;
+        }
+
         let (rect, response) = ui.allocate_at_least(desired_size, sense);
         response.widget_info(|| WidgetInfo::labeled(WidgetType::Button, galley.text()));
 
         if ui.clip_rect().intersects(rect) {
             let visuals = ui.style().interact(&response);
-            let text_pos = ui
+            let mut text_pos = ui
                 .layout()
                 .align_size_within_rect(galley.size(), rect.shrink2(button_padding))
                 .min;
 
+            if let Some(image) = image {
+                let icon_spacing = ui.spacing().icon_spacing;
+                text_pos = pos2(
+                    rect.min.x + button_padding.x + image.size().x + icon_spacing,
+                    rect.center().y - 0.5 * galley.size().y,
+                );
+            }
```

Comment:
`let text_pos = if let Some(image) = image { … } else { … };` is a nicer pattern

Reply:
I've changed to match this pattern.